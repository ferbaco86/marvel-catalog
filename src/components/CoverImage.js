import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

// const imageSize = 'detail';

const CoverImageContainerBG = styled.div`
position: relative;
width: 100%;
height: 54.5rem;
background-color: #ec1d24;
clip-path: polygon(0 0, 100% 0, 100% 72%, 0 81%);`;

const CoverImageContainer = styled.div`
position: absolute;
display: flex;
flex-direction:column;
justify-content: center;
align-items: center;
width: 100%;
height: 53rem;
background-image: url(${props => props.bgImage});
background-repeat: no-repeat;
background-size: cover;
background-position: center;
border-bottom: 3px solid #ec1d24;
clip-path: polygon(0 0, 100% 0, 100% 72%, 0 81%);
&:after {
  position: absolute;
  content: "";
  width: 100%;
  top: 0;
  bottom: 0;
  background-color: black;
  opacity: 0.5;
}`;

const Title = styled.h1`
color: white;
padding: 1rem;
font-family: 'Roboto Condensed', sans-serif;
font-weight: 700;
font-size: 5rem;
z-index: 1;
@media screen and (max-width: 800px) {
 font-size: 4rem;
  }`;

const SubTitle = styled.h3`
color: white;
padding: 1rem;
font-size: 2rem;
text-align: center;
z-index: 1`;

const CoverImage = props => {
  const { name, imageURL, subtitle } = props;
  return (
    <CoverImageContainerBG>
      <CoverImageContainer bgImage={imageURL} alt={`${name} Cover`}>
        <Title>{name}</Title>
        <SubTitle>{subtitle}</SubTitle>
      </CoverImageContainer>
    </CoverImageContainerBG>
  );
};

CoverImage.defaultProps = {
  subtitle: '',
};

CoverImage.propTypes = {
  name: PropTypes.string.isRequired,
  imageURL: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

export default CoverImage;
