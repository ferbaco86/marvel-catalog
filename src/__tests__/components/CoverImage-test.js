import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import CoverImage from '../../components/CoverImage';

const character = {
  name: 'Doctor Strange',
  url: 'www.mockurl2.com',
  subtitle: 'Multiverse Madness',
};

describe('Rendering component', () => {
  it('creates a CoverImage component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <CoverImage
              name={character.name}
              imageURL={character.url}
              subtitle={character.subtitle}
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Multiverse Madness');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <CoverImage
            name={character.name}
            imageURL={character.url}
            subtitle={character.subtitle}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
