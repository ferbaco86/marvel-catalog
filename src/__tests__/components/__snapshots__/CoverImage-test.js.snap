// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="sc-bdfBwQ hNRTSc"
>
  <div
    alt="Doctor Strange Cover"
    className="sc-gsTCUz eKsquj"
  >
    <h1
      className="sc-dlfnbm AgPsk"
    >
      Doctor Strange
    </h1>
    <h3
      className="sc-hKgILt hOQUZQ"
    >
      Multiverse Madness
    </h3>
  </div>
</div>
`;
